src/NotesApplication/Controllers/Controller.cs

```diff
@@ -1,40 +1,58 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Azure.Cosmos;
+using ACMTTU.NoteSharing.Platform.NotesApplication.Services;
+using NotesApplication;
 
-namespace ACMTTU.NoteSharing.Platform.NotesApplication.Controllers {
-    [Route ("api/notes")]
+namespace ACMTTU.NoteSharing.Platform.NotesApplication.Controllers
+{
+    [Route("api/notes")]
     [ApiController]
-    public class NotesController : PlatformBaseController {
-        private NotesMetaDataService notesMetaDataService;
+
+    public class NotesController : PlatformBaseController
+    {
+        private Container NotesContainer;
+        public NotesController(IHttpClientFactory factory, DBService db) : base(factory)
+        {
+            NotesContainer = db.container;
+        }
 
         /// <summary>
-        /// Notes Controller, Handles all of the stuff for notes
+        /// Edits given meta data assosiated with note
+        /// We will also check to see if the user has permissions to edit this note
         /// </summary>
-        /// <param name="clientFactory"> An Http Client Factory</param>
-        /// <param name="notesMetaDataService">Here is a service that I use to connect to the database</param>
-        /// <returns></returns>
-        public NotesController (IHttpClientFactory clientFactory, NotesMetaDataService notesMetaDataService) : base (clientFactory) {
-            this.notesMetaDataService = notesMetaDataService;
+        /// <param name="noteId">The id of the note to edit</param>
+        /// <param name="userId">The id of the user editting the note</param>
+        /// <param name="update">A partial object of the type Note, with the changes</param>
+        /// <returns>if note exists and user has permision to edit return Sucess else return failure.</returns>
+        [HttpPut("{noteId}/users/{userId}")]
+        public async Task<bool> EditNote(string noteId, string userId, Note update)
+        {
+            Note oldNote = await GetNote(noteId, userId);
+
+            if (oldNote.ownerId == userId) // verifies userId is allowed to modify noteId
+            {
+                await NotesContainer.ReplaceItemAsync<Note>(update, noteId);
+                return true;
+            }
+            else
+            {
+                return false;
+            }
         }
 
         /// <summary>
-        /// This is how you document code
-        /// 
-        /// Visit the microservice's endpoint and append /swagger
-        /// to see your docs in action
+        /// get a note object given the note id
         /// </summary>
-        /// <param name="id">Some sort of Id</param>
-        /// <returns>An array containing a value determined by the parameter</returns>
-        [HttpGet ("{id}")]
-        public ActionResult<string> GetValues (string id) {
-            notesMetaDataService.DoStuff ();
-            return $"Id value from URL: {id}";
+        /// <param name="noteId">The id of the not we want</param>

```

Comment:
We are missing a parameter description. We don't have the param tag for `userId` Please fix this

Reply:
@BenFischer001